plugins/out_s3/s3.c

```diff
@@ -930,13 +1000,17 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
     flb_sds_t s3_key = NULL;
     struct flb_http_client *c = NULL;
     struct flb_aws_client *s3_client;
-    struct flb_aws_header *canned_acl_header;
+    struct flb_aws_header *headers = NULL;
     char *random_alphanumeric;
     int append_random = FLB_FALSE;
     int len;
+    int ret;
+    int num_headers = 0;
     char *final_key;
     flb_sds_t uri;
     flb_sds_t tmp;
+    void *final_body;
```

Comment:
why void pointer? it should be a `char *` like `body` I think

Reply:
Yeah, but `void *` is the input type of `flb_gzip`. Although I think there is automatically conversion in C, I need to check if it will work.

Reply:
it should work. Even if it doesn't in this case you'd cast to void pointer when you make the call to flb_gzip_compress